test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -433,7 +433,7 @@ def get_system_info(self, hostname=None):
             for i in mem_info['out']:
                 if "MemTotal" in i:
                     self.system_total_ram = float(i.split()[1]) / (2**20)
-                elif "MemAvailable" in i:
+                elif "MemFree" in i:
```

Comment:
MemFree and MemAvailable are not interchangeable.

Reply:
MemAvailable is not available in every machine, but MemFree is available on every machine. I explained in the describe bug/feature.

Reply:
but they are not interchangeable - kindly explain how MemFree is doing the same thing MemAvailable was doing.

Reply:
I tested jenkins build on centos6 machine, there MemAvailable was not available.

Reply:
let me try explaining again -
I understand that MemAvailable is not there in some platforms.
However, MemFree is not same as MemAvailable. So, the calculations should be done in a different way, which is not being done in the current code changes. We need to do that.

Reply:
yes, i have write platform dependent. that is only solution for this.

Reply:
i dont think we need platform dependent solution. First understand how these 2 differ and accordingly change the calculation, if needed

Reply:
@prakashcv13 content of /proc/meminfo file
MemTotal: 3974464 kB
MemFree: 1470516 kB
Buffers: 662096 kB
Cached: 1358112 kB
SwapCached: 0 kB

centos7-
[root@centos7 pbspro]# cat /proc/meminfo
MemTotal: 7994076 kB
MemFree: 273624 kB
MemAvailable: 3237136 kB
Buffers: 263888 kB
Cached: 2841940 kB

RHEL-
MemTotal: 2047380 kB
MemFree: 563492 kB
Buffers: 679572 kB
so, I check if MemAvailable is present in first three lines then i take MemAvailable as system_ram(MemAvailable gives correct information when both are present) otherwise i take MemFree as system_ram after reading third line. reason is after first 3 lines there is no possibility of getting MemAvailable.

Reply:
get_MemAvailable = False
ram_search_count = 0
for i in mem_info['out']:
ram_search_count += 1
if "MemTotal" in i:
self.system_total_ram = float(i.split()[1]) / (220)
elif "MemFree" in i:
MemFree = float(i.split()[1]) / (220)
elif "MemAvailable" in i:
get_MemAvailable = True
MemAvailable = float(i.split()[1]) / (2**20)
if get_MemAvailable:
self.system_ram = MemAvailable
break
elif ram_search_count >= 3:
self.system_ram = MemFree
break

Reply:
there can be changes to the meminfo file, we should not rely on the order. there aren't many lines to be parsed, so we should be ok.

Reply:
should i use get_MemAvailable and remove ram_check_search?

Reply:
just change it's name to got_MemAvailable